File: RIOTNET/RiotWebProfiles/Models/LeagueInfo.cs
namespace RiotWebProfiles.Models
{
    public class LeagueInfo
    {
        public required String leagueId { get; set; }

        public required String puuid { get; set; }

        public required String queueType { get; set; }

        public required String tier { get; set; }

        public required String rank { get; set; }

        public required int leaguePoints { get; set; }

        public required int wins { get; set; }

        public required int losses { get; set; }

        public required bool hotStreak { get; set; }

        public required bool veteran { get; set; }

        public required bool freshBlood { get; set; }

        public required bool inactive { get; set; }
        
        public MiniSeries miniSeries { get; set; } //not used bcs there are promos no more

        public double winrate { get; set; }
    }
}

File: riot-profile-app/src/app/models/league.info.ts
import { MiniSeriesDTO } from "./miniseries.info";

export interface LeagueInfoDTO {
    leagueId: string;      
    puuid: string;        
    queueType: string;     
    tier: string;          
    rank: string;          
    leaguePoints: number;  
    wins: number;          
    losses: number;        
    hotStreak: boolean;    
    veteran: boolean;      
    freshBlood: boolean;   
    inactive: boolean;     
    miniSeries: MiniSeriesDTO; 
    winrate:number;
  }